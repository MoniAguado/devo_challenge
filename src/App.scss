* {
  box-sizing: border-box;
  outline: none;
}

.root {
  margin: 0;
  padding: 0;
  height: 100vh;
  background-image: url(./img/background.jpg);
  background-color: #212240;
  background-position: top left;
  background-size: cover;
  font-family: Avenir, "Avenir Next", "Segoe UI", sans-serif;
  font-weight: regular;
}

.main_app {
  position: fixed;
  top: 50%;
  left: 50%;
  display: flex;
  flex-direction: column;
  overflow-x: hidden;
  overflow-y: auto;
  max-width: 1088px;
  max-height: 648px;
  width: calc(100% - 64px);
  height: calc(100% - 64px);
  border-radius: 4px;
  background-color: #ffffff;
  opacity: 0.8;
  transform: translate(-50%, -50%);
  @media all and (max-width: 900px) {
   display: flex;
   flex-direction: column;
  }
  
}

h1 {
  position: relative;
  letter-spacing: -0.18px;
  font-size: 30px;
  line-height: 24px;
  color: #4a2c07;
}

.title {
  margin: 40px 30px 30px 40px;
}


.filters {
  margin: 20px;
  display: flex;
  justify-content: space-evenly;
}

.list {
  flex: 1;
  display: block;
  padding: 20px 60px 40px 60px;
}

.list_cities {
  padding-left: 0;
}

.cities_item {
  display: flex;
  flex-direction: row;
  padding: 20px;
  border-bottom: 1px solid rgba(74, 44, 7, 0.16);
  color: #4a2c07;
  @media all and (max-width: 900px) {
    flex-direction: column;
    margin-bottom: 20px;
   }
}


.item_city {
  width: 35%;
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
}

.item_country {
  width: 35%;
}

.item_pollution {
  width: 30%;
}

.detail {
  margin: 60px;
  padding: 0 30px 0 30px;
}

.detail_header {
  display: flex;
  flex-direction: row;
  align-items: center;
 
}

.detail_title_city {
  position: relative;
  letter-spacing: -0.18px;
  font-size: 20px;
  line-height: 24px;
  color: #4a2c07;
}

.return {
  position: absolute;
  margin-left: 430px;
  border: none;
  background: none;
  cursor: pointer;
  font-size: 16px;
  margin-left: 75%;
}

.detail_cities {
  margin-top: 20px;
}
