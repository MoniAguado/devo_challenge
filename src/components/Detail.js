import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Detail extends Component {

    constructor(props) {
        super(props)

        this.state = {
            item: {},
        }
    }

    componentDidMount() {
        const { citiesID } = this.props.routerProps.match.params;
        console.log('esto son los details', citiesID)
        const { dataList } = this.props;
        console.log('son las props de details', this.props)

        const item = dataList.find(e => e.id === citiesID)
        console.log('esto es el item', item)
        this.setState({item})
    
    }

    render() {
        const { city, country, pollution, lat, lng } = this.state.item
        console.log('csdkf', city)
        return (
             <div className="detail" >
               <button className="return">
                    <Link to="/"> Return
                    </Link>
               </button>
                <div className="cities" >
                    City: {city}
                    Country: {country}
                    Pollution: {pollution}
                    Lattitud: {lat}
                    Longitude: {lng}
                 
                </div>
                   
           </div>
        )
    }
}

export default Detail;
