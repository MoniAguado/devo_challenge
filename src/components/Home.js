import React, { Component } from 'react';
import { Link } from 'react-router-dom';

let citiesFilter = [];

class Home extends Component {

    constructor(props) {
    super(props) 
    
    this.state = {
        inputText: '',
      }
    }

    handleFilter = (event) => {
        this.setState({
          inputText: event.target.value,
        });
      }
    
    handleSortMax = () => {
        const citiesSort = this.props.dataList.sort((a, b) => b.pollution - a.pollution)
        this.setState({
            dataList: citiesSort,
        })
    }

    handleSortMin = () => {
        const citiesSort = this.props.dataList.sort((a, b) => a.pollution - b.pollution)
        this.setState({
            dataList: citiesSort,
        })
    }
    
    render() {

        const { inputText } = this.state

        citiesFilter = this.props.dataList.filter( city => {
            return  city.city
            .toUpperCase()
            .includes(this.state.inputText.toUpperCase()
          )})

        return (
            <div className="list" >
                <div className="filters">
                    <input
                        className="input"
                        placeholder=" Barcelona"
                        onChange={this.handleFilter}
                        inputText={inputText}>

                    </input>
                    <div className="buttons_box">
                        <button
                            className="sort_button"
                            onClick={this.handleSortMax}
                            >+ -
                        </button>
                        <button
                            className="sort_button"
                            onClick={this.handleSortMin}
                            >- +
                        </button>
                    </div>
                </div>
                <ul className="list_cities">
                    {citiesFilter.map( city => (
                    <Link to={`/detail/${city.id}`}>
                        <li className="cities_item" key={city.id}>
                            <div className="item_city">{city.city}</div>
                            <div className="item_country">{city.country}</div>
                            <div className="item_pollution">{city.pollution}</div>
                        </li>
                    </Link>
                    )
                    )}
                    
                </ul>
            </div>
        )
    }
}

export default Home;
