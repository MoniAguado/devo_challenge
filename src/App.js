import React, { Component} from 'react';
import './App.scss';
import getData from './api/Api';
import Home from './components/Home';
import Detail from './components/Detail';
import { Switch, Route } from 'react-router-dom';

let citiesFilter = [];

class App extends Component {

  constructor(props) {
    super(props)

    this.state = {
      dataList: [],
      inputText: '',
    }
    
  }

  componentDidMount() {
    getData().then(data => {
      this.setState({
        dataList: data,
      });
    });
  }

  handleFilter = (event) => {
    this.setState({
      inputText: event.target.value,
    });
  }

  handleSort = () => {
    const citiesSort = this.state.dataList.sort((a, b) => b.city - a.city)
    console.log('este sort que', citiesSort)
    this.setState({
      dataList: citiesSort,
    })
  }


  render() {

    console.log('Lista', this.state.dataList)

    citiesFilter = this.state.dataList.filter( city => {
      return  city.city
      .toUpperCase()
      .includes(this.state.inputText.toUpperCase()
    )})

    const { dataList, inputText, citiesSort } = this.state


    return (
      <div className="root">
        <div className="main_app">
        <h1 className="title">Help Greta!</h1> 
          <div className="filters">
            <input
              className="input"
              placeholder=" Barcelona"
              onChange={this.handleFilter}
              inputText={inputText}>

            </input>
            <button
              className="sort_button"
              onChange={this.handleSort}
              citiesSort={citiesSort}>Sort
            </button>
          </div>

        <Switch>
          
          <Route
            path="/detail/:cityID"
            render={routerProps => {
              console.log('GUARROOOOOO', dataList )
              return (
                <Detail
                  routerProps={routerProps}
                  dataList={dataList}
                 />
            )}}>
           
          </Route>
          <Route path="/">
            <Home
              citiesFilter={citiesFilter}/>
          </Route>
          
        </Switch>

        </div>
      </div>
    );
  }
}

export default App;
