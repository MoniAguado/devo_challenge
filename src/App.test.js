import React from 'react';
import App from './App';
import Home from './components/Home';
import Detail from './components/Detail';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })


describe('App', () => {
  it('App should render correctly', () => {
    const component = shallow(<App debug/>);
    
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  describe('Home', () => {
    it('home component should render correctly', () => {
      const home = shallow(<Home />);
    
      expect(home).toMatchSnapshot();
      home.unmount();
    });
  });
  // describe('Detail', () => {
  //   it('detail component should render correctly', () => {
  //     const detail = shallow(<Detail />);
    
  //     expect(detail).toMatchSnapshot();
  //   });
  // });

 
});



