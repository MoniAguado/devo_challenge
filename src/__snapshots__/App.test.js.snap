// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should render correctly 1`] = `ShallowWrapper {}`;

exports[`App Detail detail component should render correctly 1`] = `ShallowWrapper {}`;
